recipes/ceres-solver/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+                       "use_custom_blas": True,
+                       "use_eigen_sparse": True,
+                       "use_TBB": False,
+                       "use_CXX11_threads": False,
+                       "use_CXX11": False,
+                       "use_schur_specializations": True}
+    exports_sources = ["CMakeLists.txt","patches/*"]
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)       #You can check what these flags do in http://ceres-solver.org/installation.html
+            self._cmake.definitions["GFLAGS"] = self.options.use_gflags
+            self._cmake.definitions["BUILD_EXAMPLES"] = False           #Requires gflags
+            self._cmake.definitions["BUILD_TESTING"] = False            #Requires gflags
+            self._cmake.definitions["BUILD_DOCUMENTATION"] = False      #Requires python modules Sphinx and sphinx-rtd-theme
+            self._cmake.definitions["CUSTOM_BLAS"] = self.options.use_custom_blas
+            self._cmake.definitions["EIGEN_PREFER_EXPORTED_EIGEN_CMAKE_CONFIGURATION"] = False    #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["GLOG_PREFER_EXPORTED_GLOG_CMAKE_CONFIGURATION"] = False      #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["GFLAGS_PREFER_EXPORTED_GFLAGS_CMAKE_CONFIGURATION"] = False  #Set to false to Force CMake to use the conan-generated dependencies
+            self._cmake.definitions["EIGENSPARSE"] = self.options.use_eigen_sparse
+            self._cmake.definitions["SUITESPARSE"] = False  #Optional. Not sufpported right now because SuiteSparse is not part of conan-index
+            self._cmake.definitions["LAPACK"] = False       #Optional. Not supported right now because LAPACK is not part of conan-index
+            self._cmake.definitions["OPENMP"] = False
+            self._cmake.definitions["CXSPARSE"] = False     #Optional. Not supported right now because CXSSPARSE is not part of conan-index
+            self._cmake.definitions["MINIGLOG"] = not self.options.use_glog
+            self._cmake.definitions["TBB"] = self.options.use_TBB
+            self._cmake.definitions["CXX11_THREADS"] = self.options.use_CXX11_threads
+            self._cmake.definitions["CXX11"] = self.options.use_CXX11
+            self._cmake.definitions["SCHUR_SPECIALIZATIONS"] = self.options.use_schur_specializations
+            if self.settings.os == "Windows":
+                self._cmake.definitions["MSVC_USE_STATIC_CRT"] = str(self.settings.compiler.runtime) == "MT" or str(self.settings.compiler.runtime) == "MTd"
+            self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+        if self.settings.build_type == "Debug" and self.options.use_glog:
```

Comment:
Config Options can't evaluate self.options. Move all conditions that require self.options to configure(self).

Reply:
You're right. I completely missed it.